src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.h

```diff
@@ -2,20 +2,108 @@
 #define Z_EN_S_GORO_H
 
 #include "global.h"
+#include "objects/object_oF1d_map/object_oF1d_map.h"
 
-#define EN_S_GORO_GET_PARAM_F(thisx) ((thisx)->params & 0xF)
+#define EN_S_GORO_GET_MAIN_TYPE(thisx) ((thisx)->params & 0xF)
+#define EN_S_GORO_SCENEFLAG_INDEX(thisx) (((thisx)->params & 0x7F0) >> 4)
+
+#define EN_S_GORO_ROLLEDUP_YOFFSET 14.0f
+
+#define ENSGORO_ACTIONFLAG_ROLLEDUP (1 << 0)
+#define ENSGORO_ACTIONFLAG_FACEPLAYER (1 << 1)
+#define ENSGORO_ACTIONFLAG_EYESOPEN (1 << 2)
+#define ENSGORO_ACTIONFLAG_EARSCOVERED (1 << 3)
+#define ENSGORO_ACTIONFLAG_ENGAGED (1 << 4)
+#define ENSGORO_ACTIONFLAG_LASTMESSAGE (1 << 5)
+#define ENSGORO_ACTIONFLAG_GKQUIET_ACKNOWLEDGED (1 << 6)
+#define ENSGORO_ACTIONFLAG_SNOREPHASE (1 << 7)
+#define ENSGORO_ACTIONFLAG_UNK0100 (1 << 8)
+#define ENSGORO_ACTIONFLAG_HANDTAP (1 << 9)
+#define ENSGORO_ACTIONFLAG_TIRED (1 << 10)
+#define ENSGORO_ACTIONFLAG_SUPPRESS_SNORE (1 << 11)
+
+#define ENSGORO_BOMBBUYFLAG_TALKED_HUMAN (1 << 0)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORONPK (1 << 1)
+#define ENSGORO_BOMBBUYFLAG_TALKED_DEKU (1 << 2)
+#define ENSGORO_BOMBBUYFLAG_TALKED_ZORA (1 << 3)
+#define ENSGORO_BOMBBUYFLAG_TALKED_HUMAN_FINALNIGHT (1 << 4)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GOROKPK_FINALNIGHT (1 << 5)
+#define ENSGORO_BOMBBUYFLAG_TALKED_DEKU_FINALNIGHT (1 << 6)
+#define ENSGORO_BOMBBUYFLAG_TALKED_ZORA_FINALNIGHT (1 << 7)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON (1 << 8)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON_FINALNIGHT (1 << 9)
+#define ENSGORO_BOMBBUYFLAG_YESBUY (1 << 10)
 
 struct EnSGoro;
 
 typedef void (*EnSGoroActionFunc)(struct EnSGoro*, PlayState*);
 
+extern const ActorInit En_S_Goro_InitVars;
+
 typedef struct EnSGoro {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x48];
+    /* 0x0144 */ Actor* otherGoron;
```

Comment:
```suggestion
    /* 0x144 */ Actor* otherGoron;
```
Could you remove the leading 0s from these offset comments? Applies to the rest as well